src/onegov/form/validators.py

```diff
@@ -302,6 +299,47 @@ def __call__(self, form: 'Form', field: 'Field') -> None:
                 errors.append(error)
                 raise ValidationError(error)
 
+        return parsed_form
+
+    def _parse_form(self, field: 'Field',
+                    enable_indent_check: 'bool' = True) -> 'Form':
+        # XXX circular import
+        from onegov.form import parse_form
+
+        return parse_form(field.data,
+                          enable_indent_check=enable_indent_check)()
+
+
+class ValidFilterFormDefinition(ValidFormDefinition):
+    invalid_field_type = _("Invalid field type for field '{label}'. For "
+                           "filters only 'select' or 'multiple select' "
+                           "fields are allowed.")
+
+    def __call__(self, form: 'Form', field: 'Field') -> 'type[Form] | None':
+        from onegov.form.fields import MultiCheckboxField
+
+        parsed_form = super().__call__(form, field)
+        if parsed_form is None:
+            return None
+
+        # limit the definition to MultiCheckboxField, RadioField which can
+        # be used for filter definition
+        errors = None
+        for field in parsed_form._fields.values():
+            if not isinstance(field, (MultiCheckboxField, RadioField)):
+                error = field.gettext(self.invalid_field_type.format(
+                    label=field.label.text))
+                errors = form['definition'].errors
+                if not isinstance(errors, list):
+                    errors = form['definition'].process_errors
+                    assert isinstance(errors, list)
+                errors.append(error)
+
+        if errors:
+            raise ValidationError(error)
```

Comment:
If `ValidationError` contains a message it will be added to the field's error list, but since we already manually added the error so we can report multiple errors instead of just the first one, we don't need to add a new error message, we just need to tell WTForms that the validation failed.